src/main/java/org/opentripplanner/model/TripPattern.java

```diff
@@ -568,7 +568,12 @@ public Direction getDirection() {
    * trips/TripIds in the Timetable rather than the enclosing TripPattern.
    */
   public Stream<Trip> scheduledTripsAsStream() {
-    return scheduledTimetable.getTripTimes().stream().map(TripTimes::getTrip).distinct();
+    var trips = scheduledTimetable.getTripTimes().stream().map(TripTimes::getTrip);
+    var freqTrips = scheduledTimetable
+      .getFrequencyEntries()
+      .stream()
+      .map(e -> e.tripTimes.getTrip());
+    return Stream.concat(trips, freqTrips).distinct();
```

Comment:
The name of this method now become miss-leading, if I understand it right a frequency-based trip is not a "scheduled" trip.

Reply:
My first attempt had separate methods for frequency and non-frequency trips.

But if you look at the call sites you can see that all the API classes also call this method so it would fix the problem only for the geometries but not the API calls.

Frankly I don't know that the correct solution is here so lets discuss this at the dev meeting.

Reply:
As of my understanding, a frequency-based trip is considered "scheduled" if its `exact_times` is set to 1. If the `exact_times` is set to 0, the approach is pessimistic (wait time can be `headway_secs` or some kind of average). The actual behavior, after testing this commit, is respecting these hypothesis.